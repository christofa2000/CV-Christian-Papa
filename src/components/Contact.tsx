"use client";

import { motion } from "framer-motion";
import Image from "next/image";

const fade = {
  hidden: { opacity: 0, y: 12 },
  show: { opacity: 1, y: 0, transition: { duration: 0.45, ease: "easeOut" } },
};

export default function AboutMe() {
  return (
    <section id="sobre-mi" className="py-32 md:py-40 bg-neutral-900/30">
      <div className="mx-auto max-w-5xl px-4">
        {/* Texto principal */}
        <motion.div
          initial="hidden"
          whileInView="show"
          viewport={{ once: false, amount: 0.35 }}
          variants={fade}
          className="text-center"
        >
          <h2 className="text-4xl md:text-5xl font-semibold text-neutral-200 text-center">
            Sobre mí
          </h2>

          <p className="mt-6 text-lg md:text-xl text-neutral-400 leading-relaxed max-w-3xl mx-auto">
            Soy{" "}
            <span className="text-neutral-200 font-medium">
              Desarrollador Frontend
            </span>{" "}
            especializado en <span className="text-neutral-200">React</span> y{" "}
            <span className="text-neutral-200">TypeScript</span>. Creo
            experiencias web modernas, accesibles y de alto rendimiento, con
            foco en la escalabilidad y el detalle.
          </p>

          <p className="mt-5 text-lg md:text-xl text-neutral-400 leading-relaxed max-w-3xl mx-auto">
            Trabajo con <span className="text-neutral-200">Next.js</span>,{" "}
            <span className="text-neutral-200">Zustand</span>,{" "}
            <span className="text-neutral-200">React Query</span> y{" "}
            <span className="text-neutral-200">Tailwind</span>, aplicando buenas
            prácticas y tests con <span className="text-neutral-200">Jest</span>{" "}
            y <span className="text-neutral-200">React Testing Library</span>.
          </p>

          <p className="mt-5 text-lg md:text-xl text-neutral-400 leading-relaxed max-w-3xl mx-auto">
            Me definen la <span className="text-neutral-200">comunicación</span>
            , la <span className="text-neutral-200">adaptabilidad</span> y el
            trabajo en equipo, siempre buscando mejorar procesos y aprender algo
            nuevo en cada proyecto.
          </p>
        </motion.div>

        {/* Íconos del stack (skillicons.dev) */}
        <motion.div
          initial="hidden"
          whileInView="show"
          viewport={{ once: false, amount: 0.25 }}
          variants={fade}
          className="mt-14 md:mt-16 flex flex-wrap justify-center gap-4 sm:gap-5 md:gap-6"
        >
          {[
            "react",
            "nextjs",
            "typescript",
            "javascript",
            "tailwind",
            "redux",
            "prisma",
            "postgres",
            "nodejs",
            "cloudflare",
            "vercel",
            "git",
            "github",
            "figma",
            "vitest",
            "jest",
          ].map((icon) => (
            <motion.div
              key={icon}
              whileHover={{ scale: 1.12, rotate: 1 }}
              whileTap={{ scale: 0.95 }}
              transition={{ type: "spring", stiffness: 300, damping: 18 }}
              className="w-12 h-12 md:w-14 md:h-14"
            >
              <Image
                src={`https://skillicons.dev/icons?i=${icon}&theme=dark`}
                alt={icon}
                width={56}
                height={56}
                unoptimized
                className="rounded-lg"
              />
            </motion.div>
          ))}
        </motion.div>
      </div>
    </section>
  );
}
