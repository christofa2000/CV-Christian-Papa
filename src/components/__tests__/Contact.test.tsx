import { render, screen } from "@testing-library/react";
import Contact from "../Contact";

describe("Contact component", () => {
  it("renderiza el título de la sección", () => {
    render(<Contact />);
    expect(screen.getByText(/Contacto/i)).toBeInTheDocument();
  });

  it("renderiza el subtítulo descriptivo", () => {
    render(<Contact />);
    expect(
      screen.getByText(
        /¿Tienes un proyecto en mente\? Me encantaría escuchar sobre él/i
      )
    ).toBeInTheDocument();
  });

  it("renderiza los enlaces de contacto", () => {
    render(<Contact />);
    expect(screen.getByText(/Email/i)).toBeInTheDocument();
    expect(screen.getByText(/GitHub/i)).toBeInTheDocument();
    expect(screen.getByText(/LinkedIn/i)).toBeInTheDocument();
  });

  it("renderiza la información de disponibilidad", () => {
    render(<Contact />);
    expect(screen.getByText(/Disponibilidad/i)).toBeInTheDocument();
    expect(
      screen.getByText(/Actualmente disponible para proyectos freelance/i)
    ).toBeInTheDocument();
  });

  it("tiene los enlaces con los hrefs correctos", () => {
    render(<Contact />);
    const emailLink = screen.getByText(/Email/i).closest("a");
    const githubLink = screen.getByText(/GitHub/i).closest("a");
    const linkedinLink = screen.getByText(/LinkedIn/i).closest("a");

    expect(emailLink).toHaveAttribute("href", "christofa2000@gmail.com");
    expect(githubLink).toHaveAttribute(
      "href",
      "https://github.com/christofa2000"
    );
    expect(linkedinLink).toHaveAttribute(
      "href",
      "https://www.linkedin.com/in/christian-oscar-papa?utm_source=share_via&utm_content=profile&utm_medium=member_android"
    );
  });

  it("tiene los atributos ARIA correctos en los enlaces", () => {
    render(<Contact />);
    const emailLink = screen.getByText(/Email/i).closest("a");
    const githubLink = screen.getByText(/GitHub/i).closest("a");
    const linkedinLink = screen.getByText(/LinkedIn/i).closest("a");

    expect(emailLink).toHaveAttribute(
      "aria-label",
      "Enviar email a Christian Papa"
    );
    expect(githubLink).toHaveAttribute(
      "aria-label",
      "Ver perfil de GitHub de Christian Papa"
    );
    expect(linkedinLink).toHaveAttribute(
      "aria-label",
      "Ver perfil de LinkedIn de Christian Papa"
    );
  });

  it("coincide con el snapshot", () => {
    const { container } = render(<Contact />);
    expect(container).toMatchSnapshot();
  });
});
