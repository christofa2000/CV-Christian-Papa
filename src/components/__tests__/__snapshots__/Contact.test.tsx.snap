// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Contact component coincide con el snapshot 1`] = `
<div>
  <section
    class="py-20 md:py-24 bg-neutral-900/50"
    id="contacto"
  >
    <div
      class="container mx-auto px-4"
    >
      <div
        class="max-w-4xl mx-auto text-center"
      >
        <h2
          class="text-5xl md:text-6xl lg:text-7xl font-semibold text-neutral-200 mb-6"
        >
          Contacto
        </h2>
        <p
          class="text-xl md:text-2xl lg:text-3xl text-neutral-400 mb-12"
        >
          ¿Tienes un proyecto en mente? Me encantaría escuchar sobre él.
        </p>
        <div
          class="flex flex-wrap justify-center gap-6 mb-12"
        >
          <a
            aria-label="Enviar email a Christian Papa"
            class="group flex items-center gap-3 px-6 py-4 bg-neutral-800 rounded-2xl border border-white/10 shadow-lg hover:shadow-xl transition-all duration-300 focus-ring"
            href="christofa2000@gmail.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="lucide lucide-mail text-neutral-400 group-hover:text-indigo-400 transition-colors"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="16"
                rx="2"
                width="20"
                x="2"
                y="4"
              />
              <path
                d="m22 7-8.97 5.7a1.94 1.94 0 0 1-2.06 0L2 7"
              />
            </svg>
            <span
              class="font-medium text-neutral-200 group-hover:text-indigo-400 transition-colors"
            >
              Email
            </span>
          </a>
          <a
            aria-label="Ver perfil de GitHub de Christian Papa"
            class="group flex items-center gap-3 px-6 py-4 bg-neutral-800 rounded-2xl border border-white/10 shadow-lg hover:shadow-xl transition-all duration-300 focus-ring"
            href="https://github.com/christofa2000"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="lucide lucide-github text-neutral-400 group-hover:text-indigo-400 transition-colors"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
              />
              <path
                d="M9 18c-4.51 2-5-2-7-2"
              />
            </svg>
            <span
              class="font-medium text-neutral-200 group-hover:text-indigo-400 transition-colors"
            >
              GitHub
            </span>
          </a>
          <a
            aria-label="Ver perfil de LinkedIn de Christian Papa"
            class="group flex items-center gap-3 px-6 py-4 bg-neutral-800 rounded-2xl border border-white/10 shadow-lg hover:shadow-xl transition-all duration-300 focus-ring"
            href="https://www.linkedin.com/in/christian-oscar-papa?utm_source=share_via&utm_content=profile&utm_medium=member_android"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="lucide lucide-linkedin text-neutral-400 group-hover:text-indigo-400 transition-colors"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
              />
              <rect
                height="12"
                width="4"
                x="2"
                y="9"
              />
              <circle
                cx="4"
                cy="4"
                r="2"
              />
            </svg>
            <span
              class="font-medium text-neutral-200 group-hover:text-indigo-400 transition-colors"
            >
              LinkedIn
            </span>
          </a>
        </div>
        <div>
          <a
            class="inline-flex items-center gap-3 px-8 py-4 bg-indigo-600 text-white font-semibold rounded-2xl hover:bg-indigo-700 transition-colors focus-ring shadow-lg hover:shadow-xl"
            download="CV-Christian-Papa.pdf"
            href="/cv-christian-papa.pdf"
          >
            <svg
              class="lucide lucide-download"
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
              />
              <polyline
                points="7 10 12 15 17 10"
              />
              <line
                x1="12"
                x2="12"
                y1="15"
                y2="3"
              />
            </svg>
            Descargar CV
          </a>
        </div>
        <div
          class="mt-12 p-6 bg-neutral-800 rounded-2xl border border-white/10 shadow-lg"
        >
          <h3
            class="text-lg font-semibold text-neutral-200 mb-2"
          >
            Disponibilidad
          </h3>
          <p
            class="text-neutral-400"
          >
            Actualmente disponible para proyectos freelance y oportunidades de trabajo remoto. Respuesta garantizada en 24 horas.
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
